File: exercice2_1.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <sys/types.h>
#include <sys/wait.h>
#include <errno.h>

int main()
{
	pid_t pid_fils ;
	pid_fils = fork() ;
	if (pid_fils == -1)
	{
		perror("Problème dans fork\n") ;
		exit (errno) ;
	}
	else if (pid_fils == 0)
	{
		printf("je suis le fils\n") ;
	}
	else
	{
		printf("je suis le père\n") ;
		wait(NULL) ;
		printf("fils terminé\n");
	}
	return (0) ;
}

File: Python/exercice2_1.py
import os
import sys
import time

def main():
    
    pid_fils = os.fork()
    
    if pid_fils == -1:  # Erreur
        print("Problème dans fork")
        sys.exit(1)
    
    elif pid_fils == 0:  # Processus fils
        print("je suis le fils")
    
    else:  # Processus père
        print("je suis le père")
        
        # wait
        os.wait()
        print("fils terminé")

if __name__ == "__main__":
    main()